test_util/onprem.py

```diff
@@ -0,0 +1,203 @@
+""" Utilities to assist with orchestrating and testing an onprem deployment
+"""
+import copy
+import itertools
+import logging
+
+from retrying import retry
+
+from ssh.ssher import Ssher
+from test_util.helpers import ApiClientSession, Url
+
+
+log = logging.getLogger(__name__)
+
+
+def log_and_raise_if_not_ok(response):
+    if not response.ok:
+        log.error(response.content.decode())
+        response.raise_for_status()
+
+
+class OnpremCluster:
+
+    def __init__(self, ssher, masters, private_agents, public_agents, bootstrap_host):
```

Comment:
This is a good candidate to document. Please document the args and type of the args for this class.